.github/workflows/task-build-artifacts.yml

```diff
@@ -80,20 +78,73 @@ jobs:
             tests/pytests/requirements.txt
       - name: Deps checkout (node20 not supported)
         if: steps.node20.outputs.supported == 'false'
-        uses: actions/checkout@v3
-        with:
-          path: setup
-          ref: ${{ env.REF }}
-          sparse-checkout-cone-mode: false
-          sparse-checkout: |
-            .install
-            tests/pytests/requirements.txt
-      - name: Setup specific
+        run: |
+          # Checkout script logic
+          perform_checkout_ref() {
+            REPO_URL="https://github.com/RediSearch/RediSearch.git"
+            DEST_DIR="RediSearch"  # Directory to clone into
+
+            # Clone the repository to the current directory
+            git clone --recurse-submodules --depth=1 $REPO_URL .
+
+            # Configure the safe directory
+            git config --global --add safe.directory /__w/RediSearch/RediSearch
+
+            # Checkout the REF
+            git fetch origin ${{ env.REF }}
+            git checkout ${{ env.REF }}
+          }
+
+          # Execute the logic based on the detected platform
+          echo "Detected platform: ${{ inputs.container }}"
+          case "${{ inputs.container }}" in
+            ubuntu:bionic | amazonlinux:2 | alpine:3)
+              echo "Installing prerequisites for ${{ inputs.container }}..."
+
+              # Install git (platform-specific package management)
+              case "${{ inputs.container }}" in
+                ubuntu:bionic)
+                  echo "Updating package lists..."
```

Comment:
Echo messages also in other cases?

Reply:
One print before is good enough, removing this one.